common/xbps-src/shutils/common.sh

```diff
@@ -258,21 +237,26 @@ run_pkg_hooks() {
 unset_package_funcs() {
     local f
 
-    for f in $(typeset -F|grep -E '_package$'); do
-        eval unset -f $f
+    for f in $(typeset -F); do
+        case "$f" in
+        *_package)
+            set -- $f
+            eval unset -f "$3"
```

Comment:
```
$ test_package() { :; }
$ for f in $(typeset -F); do printf '[%s]\n' "$f"; done
[declare]
[-f]
[test_package]
```

the `set --` splitting doesn't work here because `$f` already contains just one word (so `$3` is always empty)

you could remove it and do just `unset -f "$f"` instead (the `eval` isn't needed)

the same applies to `get_subpkgs()` below